internal/hcsoci/resources_lcow.go

```diff
@@ -167,14 +164,13 @@ func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, res
 					},
 				},
 			}
-			vmBusGUID, err := coi.HostingSystem.AssignDevice(ctx, v)
+			vpci, err := coi.HostingSystem.AssignDevice(ctx, v)
 			if err != nil {
 				return errors.Wrapf(err, "failed to assign gpu device %s to pod %s", d.ID, coi.HostingSystem.ID())
 			}
-			resources.vpciDevices = append(resources.vpciDevices, vmBusGUID)
-
+			r.resources = append(r.resources, vpci)
 			// update device ID so the gcs knows which devices to map into the container
-			coi.Spec.Windows.Devices[i].ID = vmBusGUID
+			coi.Spec.Windows.Devices[i].ID = vpci.ID
```

Comment:
I think this is fine, but can we update the comment to something like 
`// update device ID on the spec to the assigned device's resulting vmbus guid so gcs knows which devices to map into the container`